### PR TITLE
feature/18/leave_guzzle

### PR DESCRIPTION
**Pull request by dsamblas**
First draft of the replacement with an http-client abstraction layer
WIP (For review and comment) feel free to decline :P

Get behat test pass with egeloen/http-adapter and curl

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -19,9 +19,9 @@
         "php": ">=5.3.2",
         "behat/behat": "~3.0",
         "atoum/atoum": "~1.0",
-        "guzzle/http": "~3.9",
         "symfony/property-access": "~2.4",
-        "justinrainbow/json-schema": "~1.3"
+        "justinrainbow/json-schema": "~1.3",
+        "egeloen/http-adapter": "dev-master"
```
Comment (tyx) [inline composer.json]: can we fix the deps on the last stable ?

File: src/Rest/RestApiBrowser.php [hunk 2]
```diff
@@ -48,19 +79,26 @@ public function getRequestHeaders()
...
+    }
+
+    /**
      * @param string $method
      * @param string $url
      * @param PyStringNode $body
-     * @param array $options
      */
-    public function sendRequest($method, $url, $body = null, array $options = array())
+    public function sendRequest($method, $url, $body = null)
```
Comment (tyx) [inline src/Rest/RestApiBrowser.php]: Removing `$options` means we step back on functionality. We need to think about an alternative to configure the http adapter we choose.

File: src/RestApiContext.php [hunk 3]
```diff
@@ -114,10 +127,31 @@ private function getRequest()
...
+
+        }
+        $rawHeaders .= "\n";
+        return $rawHeaders;
+    }
+
+    /**
+     * @Then print response
+     */
+    public function printResponse()
```
Comment (tyx) [inline src/RestApiContext.php]: printing only response does not mean anything for me. 
We can add `print request` step np if you have some need.  But from our side we need to keep the current behavior of `print response`.

File: src/Rest/RestApiBrowser.php [hunk 4]
```diff
@@ -22,21 +28,46 @@ class RestApiBrowser
...
-    public function __construct(HttpClient $httpClient)
+    /**
+     * @param string $base_url
+     * @param string|null $adaptor_name
+     * @throws HttpAdapterException
+     */
+    public function __construct($base_url, $adaptor_name)
     {
-        $this->httpClient = $httpClient;
+        if (is_string($adaptor_name) && HttpAdapterFactory::capable($adaptor_name))
```
Comment (tyx) [inline src/Rest/RestApiBrowser.php]: missing braces

**Comment by tyx** [comment]
Nice work, thanks !

File: src/Rest/RestApiBrowser.php (hunk 2 above)
Comment (dsamblas) [inline src/Rest/RestApiBrowser.php]: I will check

File: src/RestApiContext.php (hunk 3 above)
Comment (dsamblas) [inline src/RestApiContext.php]: Print response step should behave now as always, just two steps more added for debug and prettify test if needed.

**Comment by tyx** [comment]
Can we fixup all the commits ? Then I will merge !

**Comment by dsamblas** [comment]
Fixup done :)

**Comment by dsamblas** [comment]
mmmm you have merged #26 , which is not needed if you merge this one , so that why there are conflicts , I sorry should I had to close the #26 before

**Comment by tyx** [comment]
The #26 could be useful for 1.x user so we decided to merge it.

This one will be moved on a 2.0 version. No problem for conflict, I will manage it.

**Comment by dsamblas** [comment]
Yay :) , eager to "composer update" some  projects to see the changes :P

**Comment by tyx** [comment]
closed in favor of #29

**Pull request closed without merging**